docs/FAQ.rst

```diff
@@ -227,13 +227,7 @@ Run ``lgb.unloader(wipe = TRUE)`` in the R console, and recreate the LightGBM da
 Due to the pointers, choosing to not wipe variables will not fix the error.
 This is a known issue: `Microsoft/LightGBM#698 <https://github.com/microsoft/LightGBM/issues/698>`__.
 
-2. I used ``setinfo()``, tried to print my ``lgb.Dataset``, and now the R console froze!
```

Comment:
Please don't break references to this FAQ used at third-party sites. Just replace the answer for this question with info about that this is not actual anymore. Refer to https://github.com/microsoft/LightGBM/blob/master/docs/FAQ.rst#1-error-setup-script-specifies-an-absolute-path-when-installing-from-github-using-python-setuppy-install or https://github.com/microsoft/LightGBM/blob/master/docs/FAQ.rst#6-bagging-is-not-reproducible-when-changing-the-number-of-threads as an example.

Reply:
ah ok, yes I agree with you. Fixed in https://github.com/microsoft/LightGBM/pull/4854/commits/0770c94801c660e9a49de9d9c730fac66d1fbf67.

I checked on v3.3.0 and latest `master`, and there are no longer issues printing a Dataset (maybe as a result of work in #3016 ).

```r

library(lightgbm)

data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
print(dtrain)
dtrain$construct()
print(dtrain)
dtrain$setinfo("weight", rep(1.0, nrow(train$data)))
print(dtrain)
```